### ISSUE TITLE
flymake-mode will cause insert state cursor color get modified mid-typing.

### ISSUE THREAD

**Issue by pRot0ta1p** [meow-edit/meow#325]
To reproduce:

1. Start emacs with `emacs -Q` and add the path to meow's code to `load-path`
2. Load a theme that have enough face defined to allow meow load correctly, I use modus-vivendi.
3. require and enable meow-global-mode
4. modify the background color of meow-insert-cursor face and meow-normal-cursor face to different colors, for example:
```elisp
  (set-face-attribute 'meow-insert-cursor nil :background "Red" :inherit 'unspecified)
  (set-face-attribute 'meow-normal-cursor nil :background "Blue" :inherit 'unspecified)
```
5. start typing in scratch buffer with flymake-mode enabled, when flymake start checking for errors, and if there is one, cursor color will be changed to the normal state cursor color.

**Comment by pRot0ta1p** [comment]
I think I found the fix, `meow--update-cursor` need a bit modify

```diff
(defun meow--update-cursor ()
  "Update cursor type according to the current state.

This uses the variable meow-update-cursor-functions-alist, finds the first
item in which the car evaluates to true, and runs the cdr. The last item's car
in the list will always evaluate to true."
+  (with-current-buffer (window-buffer)
     (thread-last meow-update-cursor-functions-alist
                  (cl-remove-if-not (lambda (el) (funcall (car el))))
                  (cdar)
+                 (funcall))))
-    (funcall)))
```
I have to admit I don't know exactly why this solve my problem, maybe its because flymake create temporary buffers and meow want to set state to motion to this buffer, but something goes wrong and the actual current buffer's cursor get affected by that.

**Comment by pRot0ta1p** [comment]
Fixed with pr #327 , thanks!
